LogInterface and pickup point type

joosev opened the pull request.

Commit:
Added unit testing support

Commit:
Added log adapter to be able to log requests when needed. Also added unit some unit tests and example how to use log interface

pampyras committed:
Add pickuppoint searching by type

pampyras committed:
Add pickuppoint searching by type

joosev committed:
Merge branch 'dev' into master

joosev committed:
Merge pull request #24 from aixkalur/master

Added possibility to log requests in application that uses this library

joosev committed:
Merge pull request #29 from mijora/feature/pickup_search

Feature/pickup search

pampyras committed:
Removed draft shipment

joosev committed:
Merge pull request #31 from mijora/feature/remove_draft_support

Removed draft shipment

ikajaste-pgo commented on src/Pakettikauppa/Client.php:
```diff
@@ -204,12 +209,8 @@ private function createShipment(Shipment &$shipment, $draft = false, $language =
-            $response = $this->doPost('/prinetti/create-shipment-draft', null, $shipment_xml->asXML());
-        }
-
+        $response = $this->doPost("/prinetti/create-shipment?lang={$language}", null, $shipment_xml->asXML());
```
Draft shipment feature is deprecated, but if a draft shipment is attempted to be created, it should throw an error instead of creating a normal shipment.

joosev replied:
It thorws an exception now

pampyras committed:
Throw exception on draft shipment

pampyras committed:
Added label and cn23 count parameters

pampyras committed:
Added size

pampyras committed:
Update

joosev committed:
Merge pull request #32 from mijora/feature/remove_draft_support

Throw exception on draft shipment

nicodemuz committed:
Fix deprecation notice in SimpleXMLElement

Fixes the following PHP deprecation notice:
```
PHP Deprecated:  Return type of Pakettikauppa\SimpleXMLElement::addChild($key, $value = null, $namespace = null) should either be compatible with SimpleXMLElement::addChild(string $qualifiedName, ?string $value = null, ?string $namespace = null): ?SimpleXMLElement, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /.../vendor/pakettikauppa/api-library/src/Pakettikauppa/SimpleXMLElement.php on line 13
```

joosev committed:
Merge pull request #33 from mijora/feature/label_size_and_amount

Added label and cn23 count parameters

joosev committed:
Merge pull request #34 from nicodemuz/patch-2

Fix deprecation notice in SimpleXMLElement
